src/config/read.rs

```diff
@@ -0,0 +1,49 @@
+use tokio::fs::File;
+use tokio::io::AsyncReadExt;
+
+use crate::config::{Config, Source};
+
+pub struct ConfigReader {
+  config: Config,
+}
+
+impl ConfigReader {
+  pub fn init() -> Self {
+    Self { config: Config::default() }
+  }
+  pub async fn serialize_config(&mut self, path: &str) -> anyhow::Result<()> {
```

Comment:
```suggestion
  pub async fn read(&mut self, path: &str) -> anyhow::Result<()> {
```

Reply:
This will just create unnecessary clones and reduce performance